### ISSUE TITLE
RTK capability: (base-station) receive RTCM (1005, 1077 and 1087) corrections

### ISSUE THREAD

**Issue by AlexisTM** [Korving-F/ublox#2]
Is it possible to use this library on the UBX M8P for RTK purpose (base-station)?

configure dynamic model
get the RTCM3 corrections (1005, 1077 and 1087)
If not, do you plan on supporting it, is this project open for contributions?

**Comment by Korving-F** [comment]
Hi! Firstly; this project came mostly from my own need to read / configure a module for a weather balloon / I don't have a big background in radio/gps or the sort nor am I a great programmer. So contributions are welcome if you have some ideas ^_^

Next to that a quick glance at the [documentation](https://www.u-blox.com/sites/default/files/products/documents/u-blox8-M8_ReceiverDescrProtSpec_%28UBX-13003221%29_Public.pdf) reveals that the following messages are needed for configuration / reading in data: UBX-CFG-PRT / UBX-CFG-NAV5 / UBX-CFG-MSG / UBX-NAV-PVT /  UBX-NAV-RELPOSNED /  UBX-NAV-SAT / UBX-NAV-STATUS / UBX-INF-WARNING. (I'm probably missing something somewhere :D ) Some of these are supported, most of them only partly or not at all, but all of them should be possible to implement (if I have time next to my job / education I will definitely try :D).

tl;dr: Not possible now, should be possible with minor alterations, but I don't have much time ^_^

**Comment by AlexisTM** [comment]
It finally seems pretty simple, you can configure the messages via UBX-CFG-MSG, using:

```cpp
#define UBX_CLASS_RTCM3	0xF5

/* UBX ID for RTCM3 output messages */
/* Minimal messages for RTK: 1005, 1077 + (1087 or 1127) */
/* Reduced message size using MSM4: 1005, 1074 + (1084 or 1124)  */
#define UBX_ID_RTCM3_1005	0x05	/**< Stationary RTK reference station ARP */
#define UBX_ID_RTCM3_1074	0x4A	/**< GPS MSM4 */
#define UBX_ID_RTCM3_1077	0x4D	/**< GPS MSM7 */
#define UBX_ID_RTCM3_1084	0x54	/**< GLONASS MSM4 */
#define UBX_ID_RTCM3_1087	0x57	/**< GLONASS MSM7 */
#define UBX_ID_RTCM3_1124	0x7C	/**< BeiDou MSM4 */
#define UBX_ID_RTCM3_1127	0x7F	/**< BeiDou MSM7 */
#define UBX_ID_RTCM3_1230	0xE6	/**< GLONASS code-phase biases */
#define UBX_ID_RTCM3_4072	0xFE	/**< Reference station PVT (u-blox proprietary RTCM Message) - Used for moving baseline */

#define UBX_MSG_RTCM3_1005	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1005 << 8)
#define UBX_MSG_RTCM3_1077	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1077 << 8)
#define UBX_MSG_RTCM3_1087	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1087 << 8)
#define UBX_MSG_RTCM3_1074	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1074 << 8)
#define UBX_MSG_RTCM3_1084	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1084 << 8)
#define UBX_MSG_RTCM3_1124	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1124 << 8)
#define UBX_MSG_RTCM3_1127	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1127 << 8)
#define UBX_MSG_RTCM3_1230	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_1230 << 8)
#define UBX_MSG_RTCM3_4072	((UBX_CLASS_RTCM3) | UBX_ID_RTCM3_4072 << 8)
```

The important messages are: 1005 + 1077(GPS) + 1087(Glonass)/1127 (Beidou) + 4072 (for moving baseline)

I finally "just made it": https://github.com/AlexisTM/rtk_ros

**Comment by Korving-F** [comment]
Will close now because of lack of time; but it looks nice ^_^
